awslabs/amazon-kinesis-video-streams-ice
# ICE_API BadParams UT's

### Description of changes:

This PR is for adding additional unit tests to improve coverage, by thoroughly testing various ICE API's . Majority of the UT in this PR are for testing conditions of **Bad Parameters** and **Max Candidate Threshold**.

Also, The Formatting has been done, to make the code more readable. 

The following test cases have been added:
1. `test_iceAddServerReflexiveCandidate_BadParams`
2. `test_iceAddServerReflexiveCandidate_MaxCandidateThreshold`
3. `test_iceAddRemoteCandidate_BadParams`
4. `test_iceAddRemoteCandidate_MaxCandidateThreshold`
5. `test_iceCreateRequestForConnectivityCheck_BadParams`
6. `test_iceCreateRequestForNominatingCandidatePair_BadParams`
7. `test_iceCreateResponseForRequest_BadParams`
8. `test_iceGetLocalCandidateCount_BadParams`
9. `test_iceGetLocalCandidateCount`
10. `test_iceGetRemoteCandidateCount_BadParams`
11. `test_iceGetRemoteCandidateCount`
12. `test_iceGetCandidatePairCount_BadParams`
13. `test_iceGetCandidatePairCount`

### Test Steps

```
cmake -S test/unit-test -B build/ -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Debug -DBUILD_CLONE_SUBMODULES=ON -DCMAKE_C_FLAGS='--coverage -Wall -Wextra -Werror -DNDEBUG -DLIBRARY_LOG_LEVEL=LOG_DEBUG'
cd build 
ctest -V && make coverage
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
